src/tallies/tally_scoring.cpp

```diff
@@ -0,0 +1,2347 @@
+            }
+          }
+        } else {
+          // Skip any non-fission events
+          if (!p->fission_) continue;
+          // If there is no outgoing energy filter, than we only need to score
+          // to one bin. For the score to be 'analog', we need to score the
+          // number of particles that were banked in the fission bank. Since
+          // this was weighted by 1/keff, we multiply by keff to get the proper
+          // score. Loop over the neutrons produced from fission and check which
+          // ones are delayed. If a delayed neutron is encountered, add its
+          // contribution to the fission bank to the score.
+          if (tally.delayedgroup_filter_ != C_NONE) {
+            auto i_dg_filt = tally.filters()[tally.delayedgroup_filter_];
+            const DelayedGroupFilter& filt
+              {*dynamic_cast<DelayedGroupFilter*>(
+              model::tally_filters[i_dg_filt].get())};
+            // Tally each delayed group bin individually
+            for (auto d_bin = 0; d_bin < filt.n_bins(); ++d_bin) {
+              auto d = filt.groups()[d_bin];
+              score = simulation::keff * p->wgt_bank_ / p->n_bank_
+                * p->n_delayed_bank_[d-1] * flux;
+              score_fission_delayed_dg(i_tally, d_bin, score, score_index);
+            }
+            continue;
+          } else {
+            // Add the contribution from all delayed groups
+            auto n_delayed = std::accumulate(p->n_delayed_bank_,
+              p->n_delayed_bank_+MAX_DELAYED_GROUPS, 0);
+            score = simulation::keff * p->wgt_bank_ / p->n_bank_ * n_delayed
+              * flux;
+          }
+        }
+      } else {
+        if (i_nuclide >= 0) {
+          if (tally.delayedgroup_filter_ != C_NONE) {
+            auto i_dg_filt = tally.filters()[tally.delayedgroup_filter_];
+            const DelayedGroupFilter& filt
+              {*dynamic_cast<DelayedGroupFilter*>(
+              model::tally_filters[i_dg_filt].get())};
+            // Tally each delayed group bin individually
+            for (auto d_bin = 0; d_bin < filt.n_bins(); ++d_bin) {
+              auto d = filt.groups()[d_bin];
+              auto yield = data::nuclides[i_nuclide]
+                ->nu(E, ReactionProduct::EmissionMode::delayed, d);
+              score = p->neutron_xs_[i_nuclide].fission * yield
+                * atom_density * flux;
+              score_fission_delayed_dg(i_tally, d_bin, score, score_index);
+            }
+            continue;
+          } else {
+            // If the delayed group filter is not present, compute the score
+            // by multiplying the delayed-nu-fission macro xs by the flux
+            score = p->neutron_xs_[i_nuclide].fission
+              * data::nuclides[i_nuclide]
+              ->nu(E, ReactionProduct::EmissionMode::delayed)
+              * atom_density * flux;
+          }
+        } else {
+          if (tally.delayedgroup_filter_ != C_NONE) {
```

Comment:
Optional: It's not 100% clear from context what happens when `i_nuclide >= 0`.  Would recommend adding a comment here describing this condition and when it's expected to be executed.